packages/core/src/payment/strategies/paypal-commerce/paypal-commerce-sdk.ts

```diff
@@ -59,13 +60,131 @@ export interface PaypalButtonStyleOptions {
 export interface ButtonsOptions {
     style?: PaypalButtonStyleOptions;
     fundingSource?: string;
-    createOrder?(): Promise<string | void>; // TODO: this method should return only Promise<void>
-    onApprove?(data: ApproveDataOptions): void;
+    createOrder?(): Promise<string>;
+    onApprove?(data: ApproveDataOptions, actions: ApproveActions): void;
+    onShippingChange?(data: ShippingChangeData, actions: ApproveActions, cart: Cart): void;
     onClick?(data: ClickDataOptions, actions: ClickActions): void;
-    onCancel?(): void;
+    onCancel?(data: OnCancelData): Promise<void> | void;
```

Comment:
```suggestion
    onCancel?(data: OnCancelData): Promise<void>;
```

Reply:
replaced just with void because we don't return anything from this method